### PR TITLE
Remove `class_eval` and `class<<self` in extended

### PR DESCRIPTION
**Pull request by bquorning**
Define class method directly on the (extend-able) module, and instance method inside an `InstanceMethods` module.

@osheroff @grosser @pschambacher

### CODE REVIEW

**Comment by pschambacher** [comment]
:+1: on green

File: lib/active_record_shards/default_slave_patches.rb [hunk 1]
```diff
@@ -24,55 +24,55 @@ def #{method}_with_default_slave#{punctuation}(*args, &block)
...
-            else
-              read_columns_form = :master
-            end
-
-            on_cx_switch_block(read_columns_from, :construct_ro_scope => false) { columns_without_default_slave(*args, &block) }
-          end
-          alias_method :columns_without_default_slave, :columns
-          alias_method :columns, :columns_with_default_slave
-        end
+      base.singleton_class.send(:alias_method, :columns_without_default_slave, :columns)
```
Comment (grosser) [inline lib/active_record_shards/default_slave_patches.rb]: actually like the `class << self` more then singleton_class for doing 2-x things ... but either way is fine ...

Comment (bquorning) [inline lib/active_record_shards/default_slave_patches.rb]: What I don’t like is this pattern

```rb
module Foo
  def self.extended(base)
    base.class_eval do
      class << self
        def bar
```
much cleaner:
```rb
module Foo
  def bar
```
I’m fine with using `class << self` for those `alias_method` calls.

**Comment by bquorning** [comment]
Rebased, now using `class << self` instead of `.send`.

**Comment by bquorning** [comment]
cc @osheroff

**Comment by osheroff** [comment]
sok 👍

**Pull request merged**